pypsa/components.py

```diff
@@ -954,43 +1038,21 @@ def madd(self, class_name, names, suffix="", **kwargs):
         ...        p_max_pu=wind)
 
         """
-        if class_name not in self.components:
-            logger.error(f"Component class {class_name} not found")
-            return None
-
-        if not isinstance(names, pd.Index):
-            names = pd.Index(names)
-
-        new_names = names.astype(str) + suffix
-
-        static = {}
-        series = {}
-        for k, v in kwargs.items():
-            if isinstance(v, pd.DataFrame):
-                series[k] = v.rename(columns=lambda i: str(i) + suffix)
-            elif isinstance(v, pd.Series):
-                static[k] = v.rename(lambda i: str(i) + suffix)
```

Comment:
I don't think the processing of the index of pd.DataFrame or pd.Series type attributes to string is applied in `n.add()`. However, for consistency, it should be to align with the processing of `names` as string.

Take the example of a pd.DataFrame `df` with a range index (integer-type).

```py
n.add("Generator", df.index, p_nom=df.p_nom)
```

This will throw a value error "ValueError: Series geometry has an index which does not align with the passed names." because names is parsed to strings, but the index of df.p_nom is not. Thus, the indices don't align.